commit d48de3c145deee0c349554bf49776772d19aa2d2
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Mon Nov 27 17:14:54 2017 +0000

    cgroup: Disable cgroup "memory" by default
    
    Some Raspberry Pis have limited RAM and most users won't use the
    cgroup memory support so it is disabled by default. Enable with:
    
        cgroup_enable=memory
    
    See: https://github.com/raspberrypi/linux/issues/1950
    
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

diff --git a/kernel/cgroup/cgroup.c b/kernel/cgroup/cgroup.c
index 6e052c899cab..ea9ff74dbf9e 100644
--- a/kernel/cgroup/cgroup.c
+++ b/kernel/cgroup/cgroup.c
@@ -5278,6 +5278,8 @@ int __init cgroup_init_early(void)
 }
 
 static u16 cgroup_disable_mask __initdata;
+static u16 cgroup_enable_mask __initdata;
+static int __init cgroup_disable(char *str);
 
 /**
  * cgroup_init - cgroup initialization
@@ -5318,6 +5320,12 @@ int __init cgroup_init(void)
 
 	mutex_unlock(&cgroup_mutex);
 
+	/* Apply an implicit disable... */
+	cgroup_disable("memory");
+
+	/* ...knowing that an explicit enable will override it. */
+	cgroup_disable_mask &= ~cgroup_enable_mask;
+
 	for_each_subsys(ss, ssid) {
 		if (ss->early_init) {
 			struct cgroup_subsys_state *css =
@@ -5698,6 +5706,28 @@ static int __init cgroup_disable(char *str)
 }
 __setup("cgroup_disable=", cgroup_disable);
 
+static int __init cgroup_enable(char *str)
+{
+	struct cgroup_subsys *ss;
+	char *token;
+	int i;
+
+	while ((token = strsep(&str, ",")) != NULL) {
+		if (!*token)
+			continue;
+
+		for_each_subsys(ss, i) {
+			if (strcmp(token, ss->name) &&
+			    strcmp(token, ss->legacy_name))
+				continue;
+
+			cgroup_enable_mask |= 1 << i;
+		}
+	}
+	return 1;
+}
+__setup("cgroup_enable=", cgroup_enable);
+
 /**
  * css_tryget_online_from_dir - get corresponding css from a cgroup dentry
  * @dentry: directory dentry of interest
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1112_d48de3c145deee0c349554bf49776772d19aa2d2.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1112_d48de3c145deee0c349554bf49776772d19aa2d2.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
